lib/orangetheses/indexer.rb

```diff
@@ -91,6 +91,14 @@ def title(dc_elements)
       titles.empty? ? nil : titles.first.text
     end
 
+    def title_sort(dc_elements)
+      titles = dc_elements.select { |e| e.name == 'title' }
+      title = titles.empty? ? nil : titles.first.text
+      unless title.nil?
+        title.downcase.gsub(/\p{^Alnum}/, '').gsub(/^(?:a)|(?:an)|(?:the)/, '')
```

Comment:
gsub(/\p{^Alnum}/, '') ?

-------- Original message --------
From: Jon Stroop <notifications@github.com> 
Date: 10/29/2015  5:55 PM  (GMT-05:00) 
To: pulibrary/orangetheses <orangetheses@noreply.github.com> 
Cc: Nikitas Tampakis <nikitas.tampakis@gmail.com> 
Subject: Re: [orangetheses] Adds title_sort (#7) 

In lib/orangetheses/indexer.rb:

> @@ -91,6 +91,14 @@ def title(dc_elements)
>        titles.empty? ? nil : titles.first.text
>      end
>  
> +    def title_sort(dc_elements)
> +      titles = dc_elements.select { |e| e.name == 'title' }
> +      title = titles.empty? ? nil : titles.first.text
> +      unless title.nil?
> +        title.downcase.gsub(/\p{^Alnum}/, '').gsub(/^(?:a)|(?:an)|(?:the)/, '')
Right, but given something like this as the title:

"The Foo Bar", an Analysis

It's a lot easier to find the article if I normalize out the punctuation first.

—
Reply to this email directly or view it on GitHub.